### PR TITLE
Beta changelog

### PR DESCRIPTION
**Pull request by Manishearth**
Maintain a changelog for beta so the release team can easily include important additions in the release blog posts.

r? @phansch, @oli-obk

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,31 @@ All notable changes to this project will be documented in this file.
...
+ * New lint: [`drop_bounds`] to detect `T: Drop` bounds
+ * Rename `cyclomatic_complexity` to [`cognitive_complexity`], start work on making lint more practical for Rust code
+ * Move [`get_unwrap`] to the restriction category
+ * Improve suggestions for [`iter_cloned_collect`]
+ * Improve suggestions for [`cast_lossless`] to suggest suffixed literals
+ * Fix false positives in [`print_with_newline`] and [`write_with_newline`] pertaining to raw strings
+ * Fix false positive in [`needless_range_loop`] pertaining to structs without a `.iter()`
+ * Fix false positive in [`bool_comparison`] pertaining to non-bool types
+ * Fix false positive in [`redundant_closure`] pertaining to differences in borrows
+ * Fix false positive in [`map_or`] on non-copy types
```
Comment (phansch) [inline CHANGELOG.md]:
```suggestion
 * Fix false positive in [`option_map_unwrap_or`] on non-copy types
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,31 @@ All notable changes to this project will be documented in this file.
...
+ * Fix false positives in [`missing_const_for_fn`] pertaining to macros and trait method impls
+ * Fix false positive in [`needless_pass_by_value`] pertaining to procedural macros
+ * Fix false positive in [`needless_continue`] pertaining to loop labels
+ * Fix false positive for [`boxed_local`] pertaining to arguments moved into closures
+ * Fix false positive for [`use_self`] in nested functions
+ * Fix suggestion for [`expect_fun_call`] (https://github.com/rust-lang/rust-clippy/pull/3846)
+ * Fix suggestion for [`explicit_counter_loop`] to deal with parenthesizing range variables
+ * Fix suggestion for [`single_char_pattern`] to correctly escape single quotes
+ * Avoid triggering [`redundant_closure`] in macros
+ * ICE fixes: #3805, #3772, #3741
```
Comment (phansch) [inline CHANGELOG.md]: I think this should probably link to the PR as just the text doesn't seem very useful for readers.

Comment (Manishearth) [reply CHANGELOG.md]: Oh, I thought github would autolink

**Pull request merged**